src/main/java/io/cdap/e2e/pages/actions/CdfNameSpaceAdminActions.java

```diff
@@ -0,0 +1,119 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.e2e.pages.actions;
+
+import io.cdap.e2e.pages.locators.CdfNameSpaceAdminLocators;
+import io.cdap.e2e.pages.locators.CdfSysAdminLocators;
+import io.cdap.e2e.utils.AssertionHelper;
+import io.cdap.e2e.utils.ConstantsUtil;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.JsonUtils;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.e2e.utils.WaitHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Map;
+
+/**
+ * Represents CdfNameSpaceAdminActions
+ */
+public class CdfNameSpaceAdminActions {
+
+  private static final Logger logger = LoggerFactory.getLogger(CdfNameSpaceAdminActions.class);
+  public static CdfNameSpaceAdminLocators cdfNameSpaceAdminLocators;
+
+  static {
+    cdfNameSpaceAdminLocators = SeleniumHelper.getPropertiesLocators(
+        CdfNameSpaceAdminLocators.class);
+  }
+
+  public static void clickOnNameSpaceAdminTabs(String tabName, String nameSpaceName) {
+    ElementHelper.clickOnElement(
+        CdfNameSpaceAdminLocators.nameSpaceModules(tabName, nameSpaceName));
+  }
+
+  public static void openNamespaceAdminPage() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.namespaceAdminMenu);
+  }
+
+  public static void clickOnHamburgerMenu() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.hamburgerMenu);
+  }
+
+  public static void switchToNameSpace(String nameSpaceName) {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.switchNameSpace(nameSpaceName));
+  }
+
+  public static void createProfileForNamespace(String nameSpaceName) {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.createProfile(nameSpaceName));
+  }
+
+  public static void openNamespaceDropdown() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.namespaceDropdown);
+  }
+
+  public static void addNamespaceFromHamburgerMenu() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.addNamespace);
+  }
+
+  /**
+   * Enter KeyValue Pairs For Preference Property
+   *
+   * @param preferenceProperty @data-cy attribute value of preference Property. If
+   *                           preferenceProperty is present in
+   *                           {@link ConstantsUtil#DEFAULT_DATACY_ATTRIBUTES_FILE} then its data-cy
+   *                           is fetched from it else preferenceProperty is used as it is.
+   * @param keyValuePair       Actual json KeyValue Pairs string is fetched from
+   *                           {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} with
+   *                           keyValuePair as key
+   */
+  public static void enterKeyValuePreferences(String preferenceProperty, String keyValuePair) {
```

Comment:
The same method already exists here: https://github.com/cdapio/cdap-e2e-tests/blob/44f61d87f0e5146c2b0675cb70e86c922b479de7/src/main/java/io/cdap/e2e/pages/actions/CdfPluginPropertiesActions.java#L946

Please avoid code duplication.